airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -1575,17 +1583,47 @@ public void replaceAllConfigs(final Map<AirbyteConfig, Stream<?>> configs, final
       result.put(ConfigSchema.STANDARD_SOURCE_DEFINITION.name(),
           standardSourceDefinitionWithMetadata
               .stream()
-              .map(ConfigWithMetadata::getConfig)
-              .map(Jsons::jsonNode));
+              .map(configWitMetadata -> {
+                if (!featureFlags.hideSecretsInExport()) {
+                  return Jsons.jsonNode(configWitMetadata.getConfig());
+                }
+
+                try {
+                  final StandardSourceDefinition standardSourceDefinition = getConfig(
+                      ConfigSchema.STANDARD_SOURCE_DEFINITION,
+                      configWitMetadata.getConfigId(),
+                      StandardSourceDefinition.class);
```

Comment:
why are you doing `getConfig(...)` here? isn't this going to result in n queries? if you just call `configWithMetdata.getConfig()` doesn't that get what you need with it already in memory?

Reply:
That's true, I think that I will need that when processing the `SOURCE_CONNECTION` and `DESTINATION_CONNECTION`. I will need to use the getConfig in order to be able to get the config schema which is used to mask the secrets.